modules/playlists/src/main/java/org/opencastproject/playlists/PlaylistRestService.java

```diff
@@ -0,0 +1,600 @@
+              type = TEXT, jaxbClass = JaxbPlaylistEntry[].class, defaultValue = SAMPLE_PLAYLIST_ENTRIES_JSON)
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "Playlist updated."),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "Not authorized to perform this action")
+      })
+  public JaxbPlaylist updateEntriesAsJson(
+      @FormParam("id") String playlistId,
+      @FormParam("playlistEntries") String entriesText)
+          throws UnauthorizedException {
+    try {
+      // Map JSON to JPA
+      List<PlaylistEntry> playlistEntries = parseJsonToPlaylistEntries(entriesText);
+
+      // Persist
+      Playlist playlist = service.updateEntries(playlistId, playlistEntries);
+      return new JaxbPlaylist(playlist);
+    } catch (Exception e) {
+      throw new WebApplicationException(e);
+    }
+  }
+
+  @POST
+  @Produces(MediaType.APPLICATION_XML)
+  //  @Consumes(MediaType.APPLICATION_JSON)
+  @Path("updateEntries.xml")
+  @RestQuery(
+      name = "updateEntries",
+      description = "Updates the entries of a playlist",
+      returnDescription = "The updated playlist.",
+      restParameters = {
+          @RestParameter(name = "id", isRequired = true, description = "Playlist identifier", type = STRING),
+          @RestParameter(name = "playlistEntries", isRequired = false, description = "Playlist entries in XML format",
+              type = TEXT, jaxbClass = JaxbPlaylistEntry[].class, defaultValue = SAMPLE_PLAYLIST_ENTRIES_XML)
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "Playlist updated."),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "Not authorized to perform this action")
+      })
+  public JaxbPlaylist updateEntriesAsXml(
+      @FormParam("id") String playlistId,
+      @FormParam("playlistEntries") String entriesText)
+          throws UnauthorizedException {
+    try {
+      // Map JSON to JPA
+      List<PlaylistEntry> playlistEntries = parseXmlToPlaylistEntries(entriesText);
+
+      // Persist
+      Playlist playlist = service.updateEntries(playlistId, playlistEntries);
+      return new JaxbPlaylist(playlist);
+    } catch (Exception e) {
+      throw new WebApplicationException(e);
+    }
+  }
+
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("addEntry")
+  @RestQuery(
+      name = "addEntry",
+      description = "Add entry to playlist.",
+      returnDescription = "No content.",
+      restParameters = {
+          @RestParameter(name = "playlistId", isRequired = true, description = "Identifier of the playlist to add to",
+              type = STRING),
+          @RestParameter(name = "contentId", isRequired = false, description = "Content identifier", type = STRING),
+          @RestParameter(name = "type", isRequired = false, description = "Entry type. Enum. Valid values are EVENT,"
+              + " INACCESSIBLE.", type = STRING),
```

Comment:
It doesn't make sense to *add* an "inaccessible" entry. The "inaccessible" entry type only ever appears in the response of GET APIs. It is never stored in the DB, nor is anyone allowed to create such an entry. Maybe the `PlaylistEntryType` enum should actually be changed to not contain `INACCESSIBLE`, as that can only occur in very few places.

Reply:
But if `INACCESSIBLE` is removed from the `PlaylistEntryType` enum, the `type` field of `PlaylistEntry` cannot be set to `INACCESSIBLE` anymore? So we should we just change `type` to a String and allow whatever? Because I do not want to keep track of "`type` can be INACCESSIBLE but only in certain contexts".

Reply:
Using the type for marking an entry as inaccessible has drawbacks. You will lose the ability to tell the client which kind of information is not accessible.

Reply:
Agreed @geichelberger. Iirc the reason for not having a separate `accessible` field had something to do with future proofing external implementations of the Playlist API in case the possible types get extended to e.g. "series". I don't quite remember the exact reasoning, @LukasKalbertodt can explain it better :D

Reply:
Mhhh. So a few things:

We might add other kinds of "entries" to playlists, like series or playlists (recursively). It's unlikely that this will happen anytime soon, but I remember some people being strongly in favor of supporting recursive playlists at some point. In order to future proof the API, we already have a "type" field now so that already now, frontends can use it and deal with the "I don't even know what kind of entry this is" case. Instead of just blindly assuming today that it's always a video and then breaking in the future.

We also needed a way to represent inaccessible entries. Back then I thought: hey, we can solve both things with the "type" field! But you are right of course that this does not allow the client to know whether it's an inaccessible video or something else. I can't think of a situation where that information would be useful, though.

I'm fine with changing the API to include another field `inaccessible: bool` and always having the `type: "event" | ...` present. Slight disadvantage: clients of the API could more easily ignore the `type` field in that case, but oh well, that's their fault then. This design also helps with the "but removing `INACCESSIBLE` from `PlaylistEntryType` ..." problem. 

Should we stay with the current design and just decide clients don't get the information about the kind of inaccessible entry, then we can probably keep  `INACCESSIBLE` in `PlaylistEntryType`, but I would add a check that errors when "inaccessible" is ever tried to send to the APIs that modify entries.